### ISSUE TITLE
[Bug(?)]: Unity 2018, some timelines failed to export

### ISSUE THREAD

**Issue by Soneoylys** [AssetRipper/AssetRipper#1213]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

Windows x64

### Which Unity version did this occur on?

2018.4.7f1

### Is the game Mono or IL2Cpp?

Mono

### Describe the issue.

Some timeline failed to export with Unity 2018 (layout mismatched binary content (ArgumentOutOfRangeException)
This bug continues occurring during 0.2.x till now, on both Windows/macOS, other/older version untested.
I tried to omit this exception handler before, but the export still won't work.

If need any related files I can upload in here, but the complete game package is too big to upload.

### Relevant log output

[console output](https://github.com/AssetRipper/AssetRipper/files/14394504/Error.txt)

**Comment by Soneoylys** [comment]
Maybe is same issue as #698 
If needed, I can try to trim down the game and upload here.

**Comment by ds5678** [comment]
If those classes use `[SerializeReference]` attributes, this is indeed a duplicate of #698.

**Comment by Soneoylys** [comment]
> If those classes use `[SerializeReference]` attributes, this is indeed a duplicate of #698.

Anywhere I can confirm that?
Is that somewhere in recovered game project

**Comment by ds5678** [comment]
Load the appropriate dll assembly into ILSpy.

**Comment by Soneoylys** [comment]
> Load the appropriate dll assembly into ILSpy.

Thanks, I searched in the csharp script, but seems there's no class using [SerializeReference]

**Comment by ds5678** [comment]
> > Load the appropriate dll assembly into ILSpy.
> 
> Thanks, I searched in the csharp script, but seems there's no class using [SerializeReference]

Some Unity assemblies (correctly) do not get decompiled by AssetRipper.

**Comment by Soneoylys** [comment]
> > > Load the appropriate dll assembly into ILSpy.
> > 
> > 
> > Thanks, I searched in the csharp script, but seems there's no class using [SerializeReference]
> 
> Some Unity assemblies (correctly) do not get decompiled by AssetRipper, so searching the AssetRipper output is insufficient.

I did searched with exported script by ILSpy instead of AssetRipper result, still no luck
Or maybe I can upload trimmed game files (or managed dlls only) here?

**Comment by ds5678** [comment]
From what I can see, the issue is probably with `PropertyName exposedName`.

```cs
namespace Timeline
{
	public class TimeControlAsset : PlayableAsset, ITimelineClipAsset, IPropertyPreview
	{
		public ExposedReference<TimeControlBaseObject> Target;

		//Other content
	}
}
```
https://github.com/Unity-Technologies/UnityCsReference/blob/79868d37d65d6efb5196aaf002f97a6f87b22f97/Runtime/Export/Director/PlayableAsset.cs

https://github.com/Unity-Technologies/UnityCsReference/blob/79868d37d65d6efb5196aaf002f97a6f87b22f97/Runtime/Export/Director/ExposedReference.cs#L16-L20

**Comment by ds5678** [comment]
Seems like https://github.com/AssetRipper/AssetRipper/issues/511

**Comment by Soneoylys** [comment]
> From what I can see, the issue is probably with `PropertyName exposedName`.

Thanks for reply, I'll go check if it is this case or not

The game I've been working on uploaded to [here](https://leak.nflsga.com/internal_0.7z)

**Comment by Soneoylys** [comment]
> The game I've been working on uploaded to [here](https://leak.nflsga.com/internal_0.7z)

With the same game, AssetRipper seems also have some issue with generating outline

For example, checkPNG files in this folder: ExportedProject/Assets/Resources/common/sprites/button/

Settings I'm using:
- [Import] Skip StreamingAssets Folder --> *DISABLE*
- [Experimental] Enable Prefab Outlining --> *ENABLE*
- [Export] Sprite Export Format --> *Texture*
- [Export] Shader Export Format --> *Decompilation*
- [Export] Script Export Format --> *Decompilation*
- [Export] C# Language Version --> *C# 9*

I did bumped the editor version to 2022 LTS, but I don't think that's the problem in this case

**Comment by ds5678** [comment]
PNG files are completely unrelated to this issue. If there's a problem with them, post a new issue.

**Comment by Soneoylys** [comment]
> PNG files are completely unrelated to this issue. If there's a problem with them, post a new issue.

Thanks, I filed a new issue as #1214

**Comment by Soneoylys** [comment]
> > From what I can see, the issue is probably with `PropertyName exposedName`.
> 
> Thanks for reply, I'll go check if it is this case or not

Yes, the game script utilized `PropertyName exposedName` for timelines.

**Comment by Soneoylys** [comment]
I'm kindly asking if any progress on this issue, I tried to backflash to #511 and seems the issue closed at commit "0d45e9de6aff91fcbf20ad67862e1555a9008329", but due to this part has been reflect for a few times like #227, so
 it's sort of hard for me to debugging back those scripts to try resolve by myself.

**Comment by ds5678** [comment]
I haven't looked at it.
